src/main/java/com/minecolonies/blocks/BlockBarrel.java

```diff
@@ -0,0 +1,216 @@
+        setResistance(RESISTANCE);
+    }
+
+    public void AddItemToBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemStack, IBlockState state, BlockPos pos)
+    {
+        UseBarrel(worldIn, playerIn, itemStack, state, pos);
+    }
+
+    //whenever player right click to barrel call this.
+    public boolean UseBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemstack, IBlockState state, BlockPos pos)
+    {
+        Log.logger.info("block activated");
+
+        int barrelState = state.getValue(BARRELSTATE);
+        int fullness = fillings.getOrDefault(pos, 0);
+
+        Log.logger.info("At this moment bs= " + barrelState + " and fl=" + fullness);
+
+
+        //if statement 1
+        if (state.getValue(BARRELSTATE) == BARRELSTATE_DONE)
+        {
+            // todo: add this back in once compost exists again
+            // playerIn.inventory.addItemStackToInventory(new ItemStack(ModItems.compost, 8));
+            worldIn.setBlockState(pos, state.withProperty(BARRELSTATE, BARRELSTATE_FILLING));
+            fillings.put(pos, MIN_FULLNESS);
+            Log.logger.info("Set Blockstate to " + worldIn.getBlockState(pos));
+            return true;
+        }
+
+        if (itemstack == null)
+        {
+            return true;
+        }
+
+        Item item = itemstack.getItem();
+
+        if (item == Items.rotten_flesh && barrelState == BARRELSTATE_FILLING)
+        {
+            Log.logger.info("item Consumed");
+
+            itemstack.stackSize--;
+
+            fullness += 1;
+            if (fullness >= MAX_FULLNESS)
+            {
+                fullness = MAX_FULLNESS;
+                worldIn.setBlockState(pos, state.withProperty(BARRELSTATE, BARRELSTATE_COMPOSTING));
+            }
+            fillings.put(pos, fullness);
+            Log.logger.info("now FULLNESS = " + fullness);
+
+            return true;
+        }
+
+        return true;
+    }
+
+    public void GetItemFromBarrel(World worldIn, EntityPlayer playerIn, ItemStack itemStack, IBlockState state, BlockPos pos)
+    {
+        final int bs = state.getValue(BARRELSTATE);
+        if (bs == 2)
+        {
+            UseBarrel(worldIn, playerIn, null, state, pos);
+        }
+    }
+
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
+    {
+        return this.getDefaultState().withProperty(BARRELSTATE, meta);
+    }
+
+    @Override
+    public int getMetaFromState(final IBlockState state)
+    {
+        return state.getValue(BARRELSTATE);
+    }
+
+    /**
+     * Used to determine ambient occlusion and culling when rebuilding chunks for render.
+     *
+     * @return true
+     */
+    @Override
+    public boolean isOpaqueCube()
+    {
+        return true;
+    }
+
+    @Override
+    public void updateTick(World worldIn, BlockPos pos, IBlockState state, Random rand)
+    {
+        Log.logger.info("UpdateTick called");
+
+        int barrelState = state.getValue(BARRELSTATE);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'barrelState' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)